db/migrate/20181113190507_add_orchestration_stacks.rb

```diff
@@ -0,0 +1,22 @@
+class AddOrchestrationStacks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :orchestration_stacks do |t|
+      t.references :tenant, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref
+      t.string :name
+      t.string :description
+
+      t.timestamps
+      t.datetime :archived_at
+      t.datetime :source_created_at
+      t.datetime :source_deleted_at
+
+      t.index [:source_id, :source_ref], :unique => true
+      t.index :archived_at
+    end
+
+    add_reference :vms, :orchestration_stack, :index => true, :foreign_key => {:on_delete => :nullify}
```

Comment:
I think we'll want this on :orchestration_stack_resource, since it's the resource that maps relation to other entity. And I am not sure what'll be the best here, since resource can tie to any model in the system. Will we want foreign_key per each? (if we'd go with generic object, maybe it's not that bad) or would we go with N:M table with polymorphic relation? (but then we can't do foreign_key constraint)

Reply:
Hm okay is that different than MIQ then?  MIQ a vm belongs_to orchestration_stack

Reply:
ah, ok, I was looking at this the other way around.

Ok, we can probably keep it this way